test/filters/FreeTextFilter.test.js

```diff
@@ -0,0 +1,85 @@
+import { FreeTextFilter } from '../../src/filters/FreeTextFilter'
+import { describe, expect, test } from 'vitest'
+
+describe('FreeTextFilter', () => {
+  describe('matchesRecord()', () => {
+    test('when the record matches a term in the state: return true', () => {
+      const record = { title: { field: {}, value: 'The quick brown fox' } }
+      const freeTextFilter = new FreeTextFilter({ term: 'brown' })
+
+      expect(freeTextFilter.matchesRecord(record)).toEqual(true)
+    })
+
+    test('when the record does not match a term in the state: return false', () => {
+      const record = { title: { field: {}, value: 'The quick brown fox' } }
+      const freeTextFilter = new FreeTextFilter({ term: 'black' })
+
+      expect(freeTextFilter.matchesRecord(record)).toEqual(false)
+    })
+
+    test('when the record matches a term in the state with different capitalisation: return true', () => {
+      const record = { title: { field: {}, value: 'The quick brown fox' } }
+      const freeTextFilter = new FreeTextFilter({ term: 'BRowN' })
+
+      expect(freeTextFilter.matchesRecord(record)).toEqual(true)
+    })
+
+    test('when the record matches an integer value: return true', () => {
+      const record = { title: { field: {}, value: 17 } }
+      const freeTextFilter = new FreeTextFilter({ term: '17' })
+
+      expect(freeTextFilter.matchesRecord(record)).toEqual(true)
+    })
+
+    test('when the record does not contain the search terms spread over several fields: return false', () => {
```

Comment:
Not sure what _spread over several fields_ means here - its quite vague language. Would _when all the search terms do not appear anywhere in the record's fields_ describe what you are testing?

Reply:
I'll reword.